django_extensions/db/fields/encrypted.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+import base64
+import warnings
+
+import django
+from django import forms
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.db import models
+
+try:
+    from cryptography.fernet import Fernet
+except ImportError:
+    raise ImportError('Using an encrypted field requires the cryptography module. '
+                      'You can obtain cryptography from https://cryptography.io/.')
+
+
+class EncryptionWarning(RuntimeWarning):
+    pass
+
+
+class BaseEncryptedField(models.Field):
+
+    def __init__(self, *args, **kwargs):
+        if not getattr(settings, 'DJANGO_EXTENSIONS_ENCRYPTION_FIELDS_KEY', None):
+            raise ImproperlyConfigured('You must set the settings.DJANGO_EXTENSIONS_ENCRYPTION_FIELDS_KEY.')
+
+        self.key = getattr(settings, 'DJANGO_EXTENSIONS_ENCRYPTION_FIELDS_KEY')
+        self.fernet = Fernet(self.key)
+
+        # Encrypted size is larger than unencrypted
+        self.unencrypted_length = max_length = kwargs.get('max_length', None)
+        if max_length:
+            kwargs['max_length'] = self.calculate_crypt_max_length(max_length)
+
+        super().__init__(*args, **kwargs)
+
+    def calculate_crypt_max_length(self, unencrypted_length):
+        # TODO: Re-examine if this logic will actually make a large-enough
+        # max-length for unicode strings that have non-ascii characters in them.
+        # For PostGreSQL we might as well always use textfield since there is little
+        # difference (except for length checking) between varchar and text in PG.
+        test_bytes = ('x' * unencrypted_length).encode()
+        encrypted_data = base64.urlsafe_b64encode(self.fernet.encrypt(test_bytes)).decode()
+        return len(encrypted_data)
```

Comment:
I think allowing this seems a little odd, you may want to make the field validate the decrypted data is a specific length **before** encrypting rather than encrypting a test string and testing the encrypted length. If you still feel that it would be better to check the max length of the encrypted string, I would probably cache this value rather than run an extra encryption per `__init__`. Since it's on the model maybe I'm prematurely optimizing, but I think this would be pretty easy to do with and since it's almost what you have hear it's worth just caching the values:

```python
encrypted_lengths = {}
def calculate_crypt_max_length(unencrypted_length):
    if unencrypted_length not in encrypted_lengths:
        test_bytes = ('x' * unencrypted_length).encode()
        encrypted_data = base64.urlsafe_b64encode(self.fernet.encrypt(test_bytes)).decode()
        encrypted_lengths[unencrypted_length] = len(encrypted_data)

    return encrypted_lengths[unencrypted_length]
```

Reply:
As said in the other comment... I can well imaging that for most use-cases having a limit on the database is not a big deal... then it would be enough to check the length for validation purposes of the unencrypted value.

Also if the documentation clearly states that encrypted values are longer (prefix + whatever the encryption length expansion is) then I think that would be good enough as well... Then people are responsible themselves for dealing with this... assuming most people can live with a TextField implementation or setting something like `max_length=128, db_max_length=256` and the onces that need more should be able to easily subclass the `Field` and implement there own behavior... 

(unless somebody has a easy to implement, easy to understand, side effect free solution for this :-)